### PR TITLE
Remove JSON-based tests (as they are unreliable), along with associated helper code (cJSON, compare)

### PR DESCRIPTION
**Pull request by joedrago**
Related: #684

I don't have a strategy/design yet for a replacement set of tests that do pixel thresholds, but this design doesn't make sense with the spread of AV1 codecs / versions we have these days. We can resurrect pieces of this code in the future if we determine we need it.

### CODE REVIEW

File: tests/aviftest.c [hunk 1]
```diff
@@ -467,14 +299,10 @@ int main(int argc, char * argv[])
     // Parse cmdline
     for (int i = 1; i < argc; ++i) {
         char * arg = argv[i];
-        if (!strcmp(arg, "-g")) {
-            generate = AVIF_TRUE;
-        } else if (!strcmp(arg, "--io-only")) {
-            ioOnly = AVIF_TRUE;
+        if (!strcmp(arg, "--io-only")) {
+            // Deprecated, do nothing
```
Comment (wantehchang) [inline tests/aviftest.c:303]: Should we print a warning message that says this option is deprecated and should be removed?

File: CMakeLists.txt [hunk 2]
```diff
@@ -532,9 +532,6 @@ if(AVIF_BUILD_TESTS)
     add_executable(aviftest
         apps/shared/y4m.c
         tests/aviftest.c
-        tests/cJSON.c
```
Comment (wantehchang) [inline CMakeLists.txt:535]: Please remove the cJSON license from the LICENSE file.

**Review by wantehchang — CHANGES_REQUESTED** [review]

File: tests/aviftest.c (hunk 1 above)
Comment (joedrago) [reply tests/aviftest.c]: Done!

File: CMakeLists.txt (hunk 2 above)
Comment (joedrago) [reply CMakeLists.txt:539]: Done!

**Review by wantehchang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**